src/algebra/SparseSystem.h

```diff
@@ -96,7 +96,7 @@ class SparseSystem {
    *
    * @param n Size of the system
    */
-  SparseSystem(unsigned int n);
+  SparseSystem(int n);
```

Comment:
Why did you remove a couple of `unsigned`?

Reply:
Unsigned int's are really only used when doing something with bit values. If you set an unsigned int to a negative value then you get garbage. 

You can argue that it can be self-documenting, that `n` should be >= 0. But isn't it better to actually use a better name such as `system_size` that documents what it is rather than `n`?